doc/sphinx/documentation/developer_examples/extending_dials.rst

```diff
@@ -0,0 +1,192 @@
+   KB (dials.algorithms.scaling.model.scaling_model_ext via libtbx.dials 0.0.0)
+   array (dials.algorithms.scaling.model.scaling_model_ext via libtbx.dials 0.0.0)
+   physical (dials.algorithms.scaling.model.scaling_model_ext via libtbx.dials 0.0.0)
+
+  dxtbx.profile_model  profile models
+   gaussian_rs (dials.extensions.gaussian_rs_profile_model_ext via libtbx.dials 0.0.0)
+
+  dials.index.lattice_search_strategy  Lattice search strategies
+   low_res_spot_match (dials.algorithms.indexing.lattice_search_strategies via libtbx.dials 0.0.0)
+
+
+Adding new format classes
+=========================
+
+``dxtbx`` now discovers format classes during configuration time instead of
+at runtime. New format classes can either be added into the dxtbx/format
+directory, registered by other python packages using the
+'dxtbx.format' entry point, or installed by the user via the
+'dxtbx.install_format' command.
+
+To add a new format class to be distributed with ``dials``, please submit a pull request
+to the `dxtbx repository <https://github.com/cctbx/dxtbx>`_.
+
+To register format classes stored in ~/.dxtbx you need to run
+'dxtbx.install_format -u' whenever you add or remove format classes.
+
+As an example, let's pretend your beamline has a "reversed" rotation axis. We can create
+a new format class that correctly understands images from your beamline and instantiates
+a goniometer model with a reversed direction goniometer:
+
+.. code-block:: python
+
+  from __future__ import absolute_import, division, print_function
+
+  from dxtbx.format.FormatCBFMiniPilatus import FormatCBFMiniPilatus
+
+  class FormatCBFMiniPilatusMyBeamline(FormatCBFMiniPilatus):
+      """A class for reading mini CBF format Pilatus images for MyBeamline."""
+
+      @staticmethod
+      def understand(image_file):
+          """Check the detector serial number to check it is from MyBeamline."""
+
+          header = FormatCBFMiniPilatus.get_cbf_header(image_file)
+          for record in header.split("\n"):
+              if (
+                  "# Detector" in record
+                  and "PILATUS" in record
+                  and "S/N 42-4242" in header
+              ):
+                  return True
+          return False
+
+      def _goniometer(self):
+          """Return a model for a simple single-axis reversed direction goniometer."""
+
+          return self._goniometer_factory.single_axis_reverse()
+
+
+We can then register this format class in the ``libtbx_refresh.py`` file of our local
+``myproject`` ``cctbx`` package:
+
+.. code-block:: python
+
+  import libtbx.pkg_utils
+  libtbx.pkg_utils.define_entry_points(
+      {
+          "dxtbx.format": [
+              "FormatCBFMiniPilatusMyBeamline:FormatCBFMiniPilatus = myproject.my_format_module:FormatCBFMiniPilatusMyBeamline",
+          ],
+      }
+  )
+
+More generally, the format of an entry point for dxtbx.format is::
+
+    "FormatMyClass:FormatBaseClass1,FormatBaseClass2 = myproject.my_format_module:my"
```

Comment:
`... = myproject.my_format_module:MyFormatClass`?

Or am I reading this incorrectly?

Reply:
Not sure what happened there, now fixed.

Reply:
Cool.  You successfully unmangled my mangled suggestion!  :+1: